Add forcemerge threadpools

redcinelli opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
- Cleanup
- Docs
-->

## What does this PR do?

<!-- Mandatory
Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
-->

This Pr should enable the `force_merge` threadpool to be shipped in the monitoring event.


## Why is it important?

<!-- Mandatory
Explain here the WHY, or the rationale/motivation for the changes.
-->

We observe more and more client having the Frozen tier, and in order to have snapshot a`Force_merge` operation have to be done on each shards.
This can prove to be quite the load on the cluster, and being able to monitor it does not seems like a bad idea.

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ]

## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Superseds #123
-->

I created this PR, based off https://github.com/elastic/beats/pull/28479/files#diff-2b535cf7ec876087501fcaae889683e21f2acaab95d2274c01f647a1e1ef4e7c

## Use cases

<!-- Recommended
Explain here the different behaviors that this PR introduces or modifies in this project, user roles, environment configuration, etc.

If you are familiar with Gherkin test scenarios, we recommend its usage: https://cucumber.io/docs/gherkin/reference/
-->

## Screenshots

<!-- Optional
Add here screenshots about how the project will be changed after the PR is applied. They could be related to web pages, terminal, etc, or any other image you consider important to be shared with the team.
-->

## Logs

<!-- Recommended
Paste here output logs discovered while creating this PR, such as stack traces or integration logs, or any other output you consider important to be shared with the team.
-->

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-34711/detail/PR-34711/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-34711/detail/PR-34711/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-34711/detail/PR-34711/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-34711/detail/PR-34711/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_34711.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2023-03-01T22:11:07.247Z&rangeTo=2023-03-01T22:31:07.247Z&transactionName=BUILD+Beats%2Fbeats%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=c36b2b9c56626f7c843748f01824b9e2&transactionId=51c69aac4a83171d)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-03-01T22:21:07.247+0000


* Duration: 55 min 6 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 4377  |
| Skipped      | 887 |
| Total        | 5264   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














## :green_heart: Flaky test report
Tests succeeded.



## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.


- `/package` : Generate the packages and run the E2E tests.


- `/beats-tester` : Run the installation tests with beats-tester.


- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

redcinelli commented:
I see that 2 checks are failing with the error `Error: some files are not up-to-date. Run 'make update' then review and commit the changes.`

To fix it I do `mage update`

But it is updating quite a lot of file, I feel a review might be easier before pushing those changes.

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   beater/doc.go
	modified:   beater/metricbeat.go
	modified:   docs/fields.asciidoc
	modified:   mb/event.go
	modified:   mb/mb.go
	modified:   mb/module/options.go
	modified:   mb/testing/modules.go
	modified:   module/consul/testing.go
	modified:   module/doc.go
	modified:   module/docker/container/container.go
	modified:   module/docker/cpu/cpu.go
	modified:   module/docker/diskio/diskio.go
	modified:   module/docker/healthcheck/healthcheck.go
	modified:   module/docker/image/image.go
	modified:   module/docker/memory/memory.go
	modified:   module/docker/network/network.go
	modified:   module/docker/network_summary/config.go
	modified:   module/elasticsearch/fields.go
	modified:   module/golang/heap/data.go
	modified:   module/golang/util.go
	modified:   module/jolokia/jmx/config.go
	modified:   module/jolokia/jmx/data.go
	modified:   module/kafka/doc.go
	modified:   module/linux/rapl/rapl.go
	modified:   module/mysql/mysql.go
	modified:   module/postgresql/metricset.go
	modified:   module/postgresql/postgresql.go
	modified:   module/postgresql/testing.go
	modified:   module/redis/info/doc.go
	modified:   module/system/filesystem/doc.go
	modified:   module/system/fsstat/doc.go
	modified:   module/system/process/doc.go
	modified:   module/system/raid/blockinfo/getdev.go
	modified:   module/system/raid/blockinfo/parser.go
	modified:   module/system/service/dbus.go
	modified:   module/system/socket_summary/sockstat_linux.go
	modified:   module/system/socket_summary/sockstat_other.go
	modified:   module/windows/service/defs_service_windows.go
	modified:   module/zookeeper/mntr/mntr.go
	modified:   module/zookeeper/server/server.go
```

redcinelli commented:
As for back port of the feature.
I feel a PM might want to chip in ? Else I would not mind having it back ported as far as it can ^^

Also, applying the changes done by the `make update` might severely impact the back portability of this pr ^^

elasticmachine commented:
Pinging @elastic/integrations (Team:Platforms)

klacabane commented:
The data generated by the elasticsearch metricset can be delivered to two other places:
- [.monitoring-es-mb template](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/resources/monitoring-es-mb.json) template
- [elasticsearch integration](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/data_stream/node_stats)

Could you please add the mappings related to these fields there ?

klacabane commented on metricbeat/docs/fields.asciidoc:
```diff
@@ -29819,6 +29819,23 @@ alias to: elasticsearch.node.stats.thread_pool.bulk.rejected.count
 
 --
 
+*`node_stats.thread_pool.force_merge.queue`*::
```
I don't see these aliases defined in the mappings which is probably what is causing the build to fail.
For context, these aliases exist to make Stack Monitoring application querying easier and if we have no plan to use them there we can remove them.

redcinelli replied:
🪄  it is gone ^^

redcinelli replied:
But I believe this is not what make the build fail.
From the build logs it is `Error: some files are not up-to-date. Run 'make update'` which I am aware of but for review purposes I felt it was better not to include those changes **yet**.

redcinelli commented:
> The data generated by the elasticsearch metricset can be delivered to two other places:
> 
> * [.monitoring-es-mb template](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/resources/monitoring-es-mb.json) template
> * [elasticsearch integration](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/data_stream/node_stats)
> 
> As a follow up, could you please add the mappings related to these fields there ?

Yes, here is:

- The [Integration repo](https://github.com/elastic/integrations/pull/5508) Pr
- The [Elasticsearch repo](https://github.com/elastic/elasticsearch/pull/94469) (draft)PR

klacabane commented:
I committed the `mage update` change which only updated the relevant fields snapshot. Not sure why it would modify all those files on your end, do you have any pending local changes ?
